src/browser/BrowserShared.cpp

```diff
@@ -35,10 +36,25 @@ namespace BrowserShared
         return QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation) + "/app/" + "org.keepassxc.KeePassXC"
                + serverName;
 #elif defined(Q_OS_UNIX) && !defined(Q_OS_MACOS)
-        // Use XDG_RUNTIME_DIR instead of /tmp if it's available
-        QString path = QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation);
-        return path.isEmpty() ? QStandardPaths::writableLocation(QStandardPaths::TempLocation) + serverName
-                              : path + serverName;
+        // Use XDG_RUNTIME_DIR if available, else use /tmp.
+        QString xdgRuntimePath = QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation);
+        if (xdgRuntimePath.isEmpty())
+            return QStandardPaths::writableLocation(QStandardPaths::TempLocation) + serverName;
+
+        QDir xdgRuntimeDir = QDir(xdgRuntimePath);
```

Comment:
You don't need this variable. Just create the socketDir variable directly using `QDir(xdgRuntimePath + "/app/org.keepassxc.KeePassXC")`